### PR TITLE
[Merged by Bors] - add documentation on LogPlugin and more log usage

### LINKED ISSUE #1895: promote usage of logs rather than printing to stdout
## What problem does this solve or what need does it fill?

Examples are currently using `println!` to show what happens

## What solution would you like?

Encourage more log usage, replace all `println!` calls with logs at appropriate levels

## What alternative(s) have you considered?

just keep at that way...

## Additional context

none

### PR DESCRIPTION
**Pull request by mockersf**
Fixes #1895 

Changed most `println` to `info` in examples, some to `warn` when it was useful to differentiate from other more noisy logs.

Added doc on `LogPlugin`, how to configure it, and why (and how) you may need to disable it

### CODE REVIEW

File: crates/bevy_log/src/lib.rs [hunk 1]
```diff
@@ -16,13 +16,48 @@ use bevy_app::{AppBuilder, Plugin};
 use tracing_subscriber::fmt::{format::DefaultFields, FormattedFields};
 use tracing_subscriber::{prelude::*, registry::Registry, EnvFilter};
 
-/// Adds logging to Apps.
+/// Adds logging to Apps. This plugin is part of the `DefaultPlugins`. Adding
+/// this plugin will setup a collector appropriate to your target platform:
+/// *
```
Comment (Grindv1k) [inline crates/bevy_log/src/lib.rs:21]: Stray `*`?

File: examples/2d/many_sprites.rs [hunk 2]
```diff
@@ -90,7 +90,7 @@ fn tick(time: Res<Time>, sprites: Query<&Sprite>, mut query: Query<&mut PrintTim
         timer.0.tick(time.delta());
 
         if timer.0.just_finished() {
-            println!("Sprites: {}", sprites.iter().count(),);
+            warn!("Sprites: {}", sprites.iter().count(),);
```
Comment (Grindv1k) [inline examples/2d/many_sprites.rs:93]: Hm, not sure I see why `warn` and not `info`?

File: examples/asset/asset_loading.rs [hunk 3]
```diff
@@ -31,9 +31,9 @@ fn setup(
         // You might notice that this doesn't run! This is because assets load in parallel without
         // blocking. When an asset has loaded, it will appear in relevant Assets<T>
         // collection.
-        println!("{:?}", sphere.primitive_topology());
+        info!("{:?}", sphere.primitive_topology());
     } else {
-        println!("sphere hasn't loaded yet");
+        info!("sphere hasn't loaded yet");
```
Comment (Grindv1k) [inline examples/asset/asset_loading.rs:36]: This is nitpicking but there is some inconsistency in examples now whether capitalization is used at the start for the log messages.

E.g.

> "sphere hasn't loaded yet"

vs.

> ""Custom asset loaded: {:?}" in `custom_asset.rs`

File: examples/2d/many_sprites.rs (hunk 2 above)
Comment (mockersf) [reply examples/2d/many_sprites.rs:93]: in this example, the `LogDiagnosticsPlugin` plugin is enabled with `FrameTimeDiagnosticsPlugin`, which adds 3 lines of `INFO` log per second. As this one also logs every second, a `warn` helps it to stand out

File: crates/bevy_log/src/lib.rs (hunk 1 above)
Comment (mockersf) [reply crates/bevy_log/src/lib.rs]: forgot a commit

File: examples/2d/many_sprites.rs (hunk 2 above)
Comment (cart) [reply examples/2d/many_sprites.rs:93]: I'm not sure we should use `warn` just to increase visibility among other "noisy" logs. That encourages users to create logs in ways that don't line up with their semantics.

Comment (mockersf) [reply examples/2d/many_sprites.rs]: back to `info` it is!

File: examples/asset/asset_loading.rs (hunk 3 above)
Comment (cart) [reply examples/asset/asset_loading.rs:36]: I think starting with a capital is the right call. Feel free to follow up with another pr that makes this consistent if you feel inclined :)

**Pull request closed without merging**